qiskit/assembler/assemble_circuits.py

```diff
@@ -69,14 +69,18 @@ def _assemble_circuit(
 
     # TODO: why do we need creq_sizes and qreg_sizes in header
     # TODO: we need to rethink memory_slots as they are tied to classical bit
+    metadata = circuit.metadata
+    if metadata is None:
+        metadata = {}
     header = QobjExperimentHeader(qubit_labels=qubit_labels,
                                   n_qubits=num_qubits,
                                   qreg_sizes=qreg_sizes,
                                   clbit_labels=clbit_labels,
                                   memory_slots=memory_slots,
                                   creg_sizes=creg_sizes,
                                   name=circuit.name,
-                                  global_phase=float(circuit.global_phase))
+                                  global_phase=float(circuit.global_phase),
+                                  **metadata)
```

Comment:
It might be better to leave this as `metadata=metadata` so that we can pull out which values were added by the circuit metadata later (and not have memory slots etc get added into it). Also some metadata could potentially clash with other parameters in here.

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/5270/commits/26036dc51ee57c5d4a7dd6bd866d73bf07233cb7